File: Core/Team/src/main/java/org/team/application/commands/DeleteTeam.java
package org.team.application.commands;


import org.team.ports.dto.DeleteTeamRequest;
import org.team.ports.spi.TeamRepository;


public class DeleteTeam {
  private final TeamRepository teamRepository;

  public DeleteTeam(TeamRepository teamRepository) {
    this.teamRepository = teamRepository;
  }

  public void execute(DeleteTeamRequest request) {
    teamRepository.deleteTeam(request.teamId());
  }
}

File: frontend/src/core/application/team/command/DeleteTeam.ts
import { Command } from "src/core/application/shared/command/Command";
import { TeamCommandHandler } from "../spi/TeamCommandHandler";
import { Observable, catchError, map, of } from "rxjs";
import { Result } from "src/core/application/shared/dto/Result";



export class DeleteTeam extends Command<string, string> {

    public constructor(private commandHandler: TeamCommandHandler){super()}

    public execute(teamToDeleteId: string): Observable<Result<string>> {
        return this.commandHandler.deleteTeam(teamToDeleteId)
        .pipe(map(this.onSuccess),
        catchError(this.onError));
    }
}


